src/transformers/models/falcon_h1/modular_falcon_h1.py

```diff
@@ -617,10 +617,23 @@ def torch_forward(
 
         # 1. Gated MLP's linear projection
         input_states = apply_mask_to_padding_states(input_states, attention_mask)
+        input_states = input_states * self.ssm_in_multiplier   # ADD Mup Multipliers
         projected_states = self.in_proj(input_states)
-        gate, hidden_states_B_C, dt = projected_states.split(
-                [self.intermediate_size, self.conv_dim, self.num_heads], dim=-1
-        )
+        projected_states = projected_states * self.mup_vector  # ADD Mup Multipliers
+        d_mlp = (
+                    projected_states.shape[-1]
+                    - 2 * self.intermediate_size
+                    - 2 * self.n_groups * self.ssm_state_size
+                    - self.num_heads
+                ) // 2
+        if d_mlp > 0:
+            z0, x0, gate, hidden_states_B_C, dt = projected_states.split([
+                d_mlp, d_mlp, self.intermediate_size, self.conv_dim, self.num_heads
+            ], dim=-1)
+        else:
+            gate, hidden_states_B_C, dt = projected_states.split([
+                self.intermediate_size, self.conv_dim, self.num_heads
+            ], dim=-1)
```

Comment:
```suggestion
        d_mlp = (projected_states.shape[-1] - 2 * self.intermediate_size - 2 * self.n_groups * self.ssm_state_size-self.num_heads) // 2
        _, _, gate, hidden_states_B_C, dt = projected_states.split(
                [d_mlp, d_mlp, self.intermediate_size,  self.conv_dim, self.num_heads], dim=-1
        )
```
Is the `z0` and `x0` used? Let's remove it when it's not.

Reply:
I.e. iirc `d_mlp` should be 0 in all cases